### PR TITLE
Turn metadata sync on in isolation_update_node and isolation_update_node_lock_writes tests

### PR DESCRIPTION
**Pull request by halilozanakgul**

### CODE REVIEW

File: src/test/regress/spec/isolation_update_node_lock_writes.spec [hunk 1]
```diff
@@ -9,6 +9,10 @@ setup
 
 teardown
 {
+    UPDATE pg_dist_node SET nodeport = 57637 WHERE nodeport = 58637;
```
Comment (onderkalaci) [inline src/test/regress/spec/isolation_update_node_lock_writes.spec:12]: do we really need these?

Comment (halilozanakgul) [reply src/test/regress/spec/isolation_update_node_lock_writes.spec]: Apparently no 😄

File: src/test/regress/spec/isolation_update_node.spec [hunk 2]
```diff
@@ -8,6 +8,10 @@ setup
 
 teardown
 {
+    UPDATE pg_dist_node SET nodeport = 57637 WHERE nodeport = 58637;
```
Comment (onderkalaci) [inline src/test/regress/spec/isolation_update_node.spec:11]: what about these?

File: src/test/regress/spec/isolation_update_node.spec [hunk 3]
```diff
@@ -8,6 +8,9 @@ setup
 
 teardown
 {
+    UPDATE pg_dist_node SET metadatasynced = 't' WHERE nodeport = 57637;
```
Comment (onderkalaci) [inline src/test/regress/spec/isolation_update_node.spec:11]: Well, as you noted, given that we have to use nodes that do not exists in citus_update_node, we really need something like this.

But, doing this at the teardown seems hard to understand where we actually need to do this. Can we maybe have one explicit step like: `manually-fix-metadata` and (a) use it only when necessary (b) make it explicit what we are doing

File: src/test/regress/spec/isolation_update_node.spec [hunk 4]
```diff
@@ -103,43 +104,45 @@ step "s2-execute-prepared" {
...
 }
 
-step "s2-abort"
+
+// since we update the nodes to unexistent nodes we break metadata, so here we fix it manually
+step "s3-manually-fix-metadata"
 {
-	ABORT;
+    UPDATE pg_dist_node SET metadatasynced = 't' WHERE nodeport = 57637;
+    UPDATE pg_dist_node SET metadatasynced = 't' WHERE nodeport = 57638;
```
Comment (onderkalaci) [inline src/test/regress/spec/isolation_update_node.spec:135]: Can we move `SELECT start_metadata_sync_to_node('localhost', 57638);` here? So, we change in the coordinator an immediately sync the workers back? That feels safer / easier to follow compared having it on the tear down

**Review by onderkalaci — APPROVED** [review]
Thanks for documenting the reason for the manual metadata change

File: src/test/regress/spec/isolation_update_node_lock_writes.spec [hunk 5]
```diff
@@ -7,8 +7,11 @@ setup
     SELECT create_distributed_table('update_node', 'id');
 }
 
+// we sleep 2 seconds to let isolation test sync metadata
```
Comment (onderkalaci) [inline src/test/regress/spec/isolation_update_node_lock_writes.spec:10]: maybe note that 2 seconds is smaller than citus.metadata_sync_interval (1) for this test

**Pull request merged**